rfc/0052-sdk3-full-text-search.md

```diff
@@ -566,7 +566,275 @@ A query that matches nothing.
+            vectorSearchOptions().vectorQueryCombination(VectorQueryCombination.AND));
+
+scope.search("search_index_name", request);
+cluster.search("search_index_name", request);
+```
+
+### VectorSearch
+`VectorSearch` construction is platform-idiomatic:
+
+`VectorSearch.create(List<VectorQuery> vectorQueries, [VectorSearchOptions options])`
+
+The first parameter needs to be mandatory.
+
+`VectorSearch.create(VectorQuery vectorQuery, [VectorSearchOptions options])` can be added to make it easier to send a single `VectorQuery`.
+This is left optional to better support SDKs without overloads.
+
+`VectorSearchOptions` will currently support just one option:
+
+* `vectorQueryCombination` (`VectorQueryCombination`).  Will be sent in the top-level JSON payload as `knn_operator` (string) as either `"and"` or `"or"`.  We name the field differently in the SDK to a) improve readability and b) be able to better support possible future forms of vector search such as ANN.  If not set by the user, no value is sent.  It controls how elements in the `knn` array are combined.
+
+```
+enum VectorQueryCombination {
+    AND,
+    OR
+}
+```
+
+`VectorSearch` does not extend the `SearchQuery` interface, which is now reserved for traditional FTS queries.
+
+### VectorQuery
+`VectorQuery` creation is platform-idiomatic:
+
+`VectorQuery.create(String vectorFieldName, float[] vector)`
+
+Both parameters need to be mandatory.  They are sent in the JSON as "field" (string) and "vector" (number array) fields respectively. 
+
+The current size limit for the vector is 2048 elements, but this will not be enforced on the SDK side.
+The vector must be non-empty though, and if not the SDK will raise `InvalidArgument`. 
+
+It will also support the following parameters, exposed in the same way as the SDK exposes traditional FTS query parameters.
+In some SDKs this is as fluent-style methods on `VectorQuery` itself, but a `VectorQueryOptions` block as an optional parameter on `Vector.create()` would be more SDK3-idiomatic.
+The SDK should follow its existing convention for FTS parameters.
+
+* `uint32 numCandidates`.  Sent as a `k` number field in the JSON.  We name the field differently in the SDK for the same reason as `knn_operator`.  If not set by the user, FTS PM requests that the SDK send a default value of 3.  It controls how many results are returned and must be >= 1.  If < 1 the SDK will raise `InvalidArgument`.
+* `float boost`.  Sent as a `boost` number field in the JSON.  If not set, the field is not sent.
+
+`VectorQuery` does not extend the `SearchQuery` interface, which is now reserved for traditional FTS queries.
+
+### SearchRequest
+`SearchRequest` creation is platform-idiomatic.  There need to be two contruction options, allowing it to be created from either a `VectorSearch` or a `SearchQuery`.
+
+```
+SearchRequest.vectorSearch(VectorSearch vectorSearch, [SearchOptions searchOptions])
```

Comment:
These methods appear to have a `SearchOptions` second parameter, but I think above the `SearchOptions` are indicated as being exposed as part of the `cluster.search()` / `scope.search()`?

Reply:
Yep you're right, this one's a bug and SearchOptions should not be here.